packages/@lwc/integration-karma/test/directive-external/index.spec.js

```diff
@@ -0,0 +1,76 @@
+import { createElement } from 'lwc';
+import { isNativeShadowRootInstance } from 'test-utils';
+
+import Test from 'x/test';
+
+describe('lwc:external', () => {
+    let test;
+
+    beforeEach(() => {
+        test = createElement('x-test', { is: Test });
+        document.body.appendChild(test);
+    });
+
+    describe('when not upgraded', () => {
+        it('should set only attributes on mount', () => {
+            return Promise.resolve().then(() => {
+                const elm = test.shadowRoot.querySelector('foo-upgrade-never');
+                expect(elm.getAttribute('foo')).toBe('default');
+                expect(elm.foo).toBeUndefined();
+            });
+        });
+
+        it('should set only attributes on update', () => {
+            test.value = 'apple';
+            return Promise.resolve().then(() => {
+                const elm = test.shadowRoot.querySelector('foo-upgrade-never');
+                expect(elm.getAttribute('foo')).toBe('apple');
+                expect(elm.foo).toBeUndefined();
+            });
+        });
+    });
+
+    if (!process.env.COMPAT) {
+        describe('after upgrading', () => {
+            // This test was not broken up into smaller ones with individual assertions because we
+            // cannot manage the order in which different tests run. Order is important because once you
+            // define a custom element, there is no way to undefine it.
+            it('should set property instead of attribute', async () => {
+                const elm = test.shadowRoot.querySelector('foo-upgrade-after');
+
+                test.value = 'sake';
+                await Promise.resolve();
```

Comment:
I had assumed there was some reason we weren't using `await` in our Karma tests (Babel issue?), but it looks like we're actually already using it in a few places. TIL.